File: java_src/foam/lib/parse/Literal.java
package foam.lib.parse;

public class Literal implements Parser {
  private String string;
  private String value;

  public Literal(String s) {
    this(s, s);
  }

  public Literal(String s, String v) {
    string = s;
    value = v;
  }

  public PStream parse(PStream ps, ParserContext x) {
    for ( int i = 0 ; i < string.length() ; i++ ) {
      if ( ! ps.valid() ||
           ps.head() != string.charAt(i) ) {
        return null;
      }

      ps = ps.tail();
    }

    return ps.setValue(value);
  }
}

File: src/foam/swift/parse/parser/Literal.js
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.CLASS({
  package: 'foam.swift.parse.parser',
  name: 'Literal',
  implements: ['foam.swift.parse.parser.Parser'],
  properties: [
    {
      class: 'String',
      name: 'string',
    },
    {
      name: 'value',
      swiftExpressionArgs: ['string'],
      swiftExpression: 'return string'
    },
  ],
  methods: [
    {
      name: 'parse',
      swiftCode: function() {/*
var ps = ps
for i in 0..<string.count {
  if !ps.valid() || ps.head() != string.char(at: i) {
    return nil
  }
  ps = ps.tail()
}
return ps.setValue(self.value)
      */},
    },
  ]
});
